src/cls/rgw/cls_rgw.cc

```diff
@@ -2375,6 +2382,10 @@ static int list_instance_entries(cls_method_context_t hctx, const string& name,
     entry.data = iter->second;
 
     if (!filter.empty() && entry.idx.compare(0, filter.size(), filter) != 0) {
+      /* we are skipping the rest of the entries */
+      if (pmore) {
+	*pmore = false;
+      }
       return count;
     }
 
```

Comment:
@oritwas missing another case in each of `list_instance_entries()` and `list_olh_entries()`